Makefile

```diff
@@ -1,50 +1,87 @@
+# Convenience data so we can run the following and include
+# sources from three directories deep.
+#
+# Example:
+# $(subst dir, emily/cdk/lib, $(THREE_DIRS_DEEP))
+# becomes
+# emily/cdk/lib/*  emily/cdk/lib/*/*  emily/cdk/lib/*/*/*
+THREE_DIRS_DEEP := dir/* dir/*/* dir/*/*/*
+TWO_DIRS_DEEP := dir/* dir/*/*
+
+AUTOGENERATED_EMILY_CLIENT := .generated-sources/emily/src/lib.rs
+EMILY_LAMBDA_BINARY := target/lambda/emily-lambda/bootstrap.zip
+EMILY_CDK_TEMPLATE := emily/cdk/cdk.out/EmilyStack.template.json
+EMILY_DOCKER_COMPOSE := docker-compose.emily.yml
+INSTALL_TARGET := pnpm-lock.yaml
+
+# Don't use the install target here so you can rerun install without
+# Makefile complaints.
 install:
 	pnpm install
+	touch pnpm-lock.yaml
 
-build: install | emily-client-source
+build: $(INSTALL_TARGET) $(AUTOGENERATED_EMILY_CLIENT)
 	cargo build
 	pnpm --recursive build
 
-test: install | emily-client-source
+test: $(INSTALL_TARGET) $(AUTOGENERATED_EMILY_CLIENT)
 	cargo test
 	pnpm --recursive test
 
-integration-test: install | emily-client-source
+integration-test: $(INSTALL_TARGET) $(AUTOGENERATED_EMILY_CLIENT)
 	docker compose --file docker-compose.test.yml up --detach
 	cargo test --all-features
 	pnpm --recursive test
 	docker compose --file docker-compose.test.yml down
 
-lint: install | emily-client-source
+lint: $(INSTALL_TARGET) $(AUTOGENERATED_EMILY_CLIENT)
 	cargo clippy -- -D warnings
 	pnpm --recursive run lint
 
 clean:
 	cargo clean
 	pnpm --recursive clean
+	rm -rf devenv/dynamodb/data/*
+	@touch package.json
+
+.PHONY: install build test integration-test lint clean
+
+$(INSTALL_TARGET): $(wildcard package* */package* */*/package*)
+	pnpm install
+	touch pnpm-lock.yaml
 
 # Emily API
 # ----------------------------------------------------
 
-EMILY_PATH=emily
+EMILY_DEVENV_PATH=emily/devenv
+EMILY_LAMBDA_PATH=emily/lambda
 EMILY_API_PROJECT_NAME=emily-api
 EMILY_CDK_PROJECT_NAME=emily-cdk
 
-emily-cdk: emily-client | emily-lambda
-	pnpm --filter $(EMILY_CDK_PROJECT_NAME) run build
+# Launches Emily dev environment.
+emily-integration-test: devenv $(EMILY_LAMBDA_BINARY) $(EMILY_CDK_TEMPLATE) $(EMILY_DOCKER_COMPOSE)
```

Comment:
I tried running this locally, but my dynamodb container seems to get stuck on this error

```
Attaching to apigateway-1, aws-setup-1, dynamodb-1
dynamodb-1    | Initializing DynamoDB Local with the following configuration:
dynamodb-1    | Port:	8000
dynamodb-1    | InMemory:	false
dynamodb-1    | Version:	2.4.0
dynamodb-1    | DbPath:	./data
dynamodb-1    | SharedDb:	true
dynamodb-1    | shouldDelayTransientStatuses:	false
dynamodb-1    | CorsParams:	null
dynamodb-1    |
dynamodb-1    | May 16, 2024 5:30:14 PM com.almworks.sqlite4java.Internal log
dynamodb-1    | WARNING: [sqlite] cannot open DB[1]: com.almworks.sqlite4java.SQLiteException: [14] unable to open database file
dynamodb-1    | May 16, 2024 5:30:14 PM com.almworks.sqlite4java.Internal log
dynamodb-1    | SEVERE: [sqlite] SQLiteQueue[shared-local-instance.db]: error running job queue
dynamodb-1    | com.almworks.sqlite4java.SQLiteException: [14] unable to open database file
dynamodb-1    | 	at com.almworks.sqlite4java.SQLiteConnection.open0(SQLiteConnection.java:1480)
dynamodb-1    | 	at com.almworks.sqlite4java.SQLiteConnection.open(SQLiteConnection.java:282)
dynamodb-1    | 	at com.almworks.sqlite4java.SQLiteConnection.open(SQLiteConnection.java:293)
dynamodb-1    | 	at com.almworks.sqlite4java.SQLiteQueue.openConnection(SQLiteQueue.java:464)
dynamodb-1    | 	at com.almworks.sqlite4java.SQLiteQueue.queueFunction(SQLiteQueue.java:641)
dynamodb-1    | 	at com.almworks.sqlite4java.SQLiteQueue.runQueue(SQLiteQueue.java:623)
dynamodb-1    | 	at com.almworks.sqlite4java.SQLiteQueue.access$000(SQLiteQueue.java:77)
dynamodb-1    | 	at com.almworks.sqlite4java.SQLiteQueue$1.run(SQLiteQueue.java:205)
dynamodb-1    | 	at java.base/java.lang.Thread.run(Thread.java:840)
```

Anything you recognize?

Reply:
Yep, that only happens to suckers on Arch linux, sorry man

Reply:
Trying to fix it and planned to put that in a follow on PR for other architectures #74, there's another error you'll uniquely get too once you fix that one. You can remove the shared database file and it'll be okay but then you're likely to get a docker issue from the SAM container.

@djordon I'm curious if you get this, I assume you won't.

Reply:
I should have noted that actually, I intended for this to be enough but it's very much not and not even accurate, my bad:
> Note that this may still not work for some machines with weird docker setups.

Reply:
Hahah, my machine is definitely weird but docker is one of the things that I have working properly :sweat_smile:

Reply:
> Trying to fix it and planned to put that in a follow on PR for other architectures https://github.com/stacks-network/sbtc/issues/74, there's another error you'll uniquely get too once you fix that one. You can remove the shared database file and it'll be okay but then you're likely to get a docker issue from the SAM container.

Okay yeah that sucks. And you're running this on a non-Arch machine?

Reply:
Yes but don't worry about it let me fix it, I don't plan on leaving you out to dry

Reply:
I'll add it to this PR, probably better I don't segment it

Reply:
Yeah, I worked perfectly for me 😄

Reply:
Sweet, thanks! :pray:

Reply:
Let me know when I should try again.

Reply:
It'll be sometime tomorrow, today is GH project board day

Reply:
I seem to have better luck with this now. However `service-test` seems to fail for me on both `curl` and `jq` not found.

Reply:
Tried again and now it worked.